main/src/io/github/iltotore/iron/constraint/operators.scala

```diff
@@ -0,0 +1,45 @@
+package io.github.iltotore.iron.constraint
+
+import io.github.iltotore.iron.{==>, Constraint, Implication}
+
+import scala.compiletime.ops.boolean
+import scala.compiletime.summonInline
+
+/**
+ * Constraint operators (e.g [[operators.Not]]...).
+ */
+object operators:
```

Comment:
I mean, I think `Not`, union and intersection should be in the same package. I would rather prefer one of these suggestions:
- Move `Not`, union and intersection in `operator`
- Move `Not` with union and intersection in `Constraint`'s companion
- Keep union and intersection in `Constraint`s companion and `Not` in `any`. (Althrough I think the two above are better solutions)

Reply:
I feel all your points need some clarification 😄 

> Move Not, union and intersection in operator

What do you mean with moving union and intersection? These are language types.

> Move Not with union and intersection in Constraint's companion

This does not make sense to me, as commented [here](https://github.com/Iltotore/iron/pull/70#discussion_r1039509353).

> Keep union and intersection in Constraints companion and Not in any. (Althrough I think the two above are better solutions)

There are only instances under the `Constraint` object, not type definitions, so not sure what you mean.

Reply:
After re-reading all your suggestions, I think you're confusing customs types (`Not`) with language constructs (`&` and `|`). Here's what would happen if we follow along any of these suggestions. 

> Move Not, union and intersection in operator

If you mean move union and intersection typeclass instances to the `constraint.operator` package, users will need to import `iron.{given, *}` to get union and intersection types working.

> Move Not with union and intersection in Constraint's companion

Users would need to import `iron.Constraint.Not` to use `Not`, losing consistency with other constraints.

> Keep union and intersection in Constraints companion and Not in any. (Althrough I think the two above are better solutions)

This is how it was until the previous suggested change, which I don't think it's that bad, but moving the `Not` type under `constraint.operator` made sense too.

Reply:
> After re-reading all your suggestions, I think you're confusing customs types (Not) with language constructs (& and |). Here's what would happen if we follow along any of these suggestions.

Actually not confusing but you got my point of view: I tend to prefer to put Not/union/intersection together because the constraints `C1 | C2` behaves like the boolean "or"and the constraint `C1 & C2` behave like "and", just like `Not` is analoguous to the boolean "not".

Yes, internally union and intersection constraints are really different from `Not` but from a user perspective, I think it makes sense to consider them similar.

> If you mean move union and intersection typeclass instances to the constraint.operator package, users will need to import iron.{given, *} to get union and intersection types working.

> Users would need to import iron.Constraint.Not to use Not, losing consistency with other constraints.

Fair point. I didn't think about these two issues (damn orphan instances!). Then it seems justified to keep `Not` in `any` like you originally did.

Reply:
No problems, I understand how `And`, `Not`, and `Or` are related, but didn't make sense on the implementation side :)

Take `Functor` from Cats, for instance. The type is defined in Cats, and the instances for language constructs `Functor[List]`, `Functor[Option]`, and so on are defined within the `Functor` companion object. 

In Iron's case, it's the `Constraint` type and the instances for language constructs `&` and `|` are defined within the companion object. 

On the other hand, custom types such as `cats.data.Kleisli`, define the `Functor` and other typeclass instances within the `Kleisli` [companion object](https://github.com/typelevel/cats/blob/main/core/src/main/scala/cats/data/Kleisli.scala#L162). This is the same case for Iron's `Not`.  

> Hope you understand my point of view now and finally your first arrangement was probably the best sweat_smile

All good :) So what do we do, `any.Not` or `operator.Not`?

Reply:
I think `any` is fine since we probably wouldnt have a lot of "operators" like `Not`.

Which one do you prefer ?

Reply:
`any` is fine with me 👍🏽

Reply:
Then I will merge this PR once the refactor is done. 👍

Reply:
Done.